contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--4.0.0.sql

```diff
@@ -72,7 +72,11 @@ BEGIN
             RAISE EXCEPTION 'Role "bbf_role_admin" already exists.';
     ELSE
         EXECUTE format('CREATE ROLE bbf_role_admin WITH CREATEDB CREATEROLE INHERIT PASSWORD NULL');
+
+        SET babelfishpg_tsql.disable_storing_init_privs = TRUE;
```

Comment:
this is error prone, we can leak it in case of errors. Also, someone can add new grant/revoke without adding these GUC barriers. Is it possible to put check in grant/revoke code? we need to check if we are doing grant/revoke as part of bbf tsql extension create/upgrade?

Reply:
> this is error prone, we can leak it in case of errors.

I wrongly believed that upgrade script is executed in single transaction, and if any error occur it will rollback the whole transaction. I missed the fact that not all errors can abort the transaction. I will get back to the C language function/procedure approach.

> Also, someone can add new grant/revoke without adding these GUC barriers. Is it possible to put check in grant/revoke code? we need to check if we are doing grant/revoke as part of bbf tsql extension create/upgrade?

Sure, Let me add it.